src/viam/sdk/resource/resource_server_base.hpp

```diff
@@ -1,14 +1,15 @@
 #pragma once
 
 #include <viam/sdk/resource/resource_manager.hpp>
-#include <viam/sdk/rpc/server.hpp>
 
 namespace viam {
 namespace sdk {
 
 class ResourceServer {
    public:
+    virtual API api() const = 0;
```

Comment:
Why do `ResourceServer`'s need this now?

Reply:
My initial thinking was: when initializing the `server` we store all supported `ResourceServer` instances, but we want to be able to look them up so we store them by `API`. This is how we get the API from the `ResourceRegistration`.

However! I now realize that the key in `ResourceRegistration` is an `API` already 🤦 so I think we can get rid of this (and do away with the question of the `Robot` `API`) entirely.